viewing_party/party.py

```diff
@@ -1,23 +1,194 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
+def create_movie(movie_title, genre, rating):
+
+    movie= {}
+
+    if movie_title and genre and rating:
+    
+        movie["title"] = movie_title
+        movie["genre"] = genre
+        movie["rating"] = rating          
+        return movie
+
+    return None
+
+def add_to_watched(user_data, movie):
+
+    # Create a copy of user data to avoid modifying original
+    updated_user_data = {
+            "watched": user_data["watched"][:]}
+
+    updated_user_data["watched"].append(movie)
+    return updated_user_data
+
 
-def create_movie(title, genre, rating):
-    pass
+def add_to_watchlist(user_data, movie):
+
+    # Create a copy of user data to avoid modifying original
+    updated_user_data = {
+            "watchlist": user_data["watchlist"][:],
+        } 
+    updated_user_data["watchlist"].append(movie)
+    return updated_user_data
+
+
+def watch_movie(user_data, title):
+
+    # Create a copy of user data to avoid modifying original 
+    updated_user_data = {
+            "watchlist": user_data["watchlist"][:],
+            "watched": user_data["watched"][:]
+        } 
+    
+    # Check if movie with the title is in watchlist 
+    # and add it to the watched list
+    for movie in updated_user_data["watchlist"]:
+        if movie["title"] == title:
+            updated_user_data["watchlist"].remove(movie)
+            updated_user_data["watched"].append(movie)         
+    return updated_user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+
+    # User data is a dict which contains "watched" key
+    count = 0.0
+    sum = 0.0
+
+    if len(user_data["watched"]) == 0:
+        return sum
+    
+    # Count and sum rating values
+    for movie in user_data["watched"]:
+        if movie["rating"]:
+            count += 1

```

Comment:
Do we need to keep count of the number of movies a user has watched or can we just use a built in function provided to us by python instead in this function? 👀